Charon/filetypes/GCodeSocket.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2018 Ultimaker B.V.
+# libCharon is released under the terms of the LGPLv3 or higher.
+#
+# This class is used to read GCode stream that are served
+# dynamically over a TCP connection.
+
+import socket
+import struct
+from io import BytesIO, SEEK_SET, SEEK_CUR
+
+from typing import Any, Dict, IO, Optional, List
+
+from Charon.filetypes.GCodeFile import GCodeFile
+from urllib.parse import urlparse
+
+
+class SocketFileStream(BytesIO):
+    def __init__(self, sock_object: socket.socket):
```

Comment:
Missing return type. 

If a function isn't fully typed, the function is _not_ checked by mypy